src/js/modules/infragistics.ui.editors.js

```diff
@@ -10289,13 +10308,14 @@
 			if (!date) {
 				date = this._setNewDateMidnight();
 			}
-			if (newPeriod !== period) {
-				this._setDateField(periodName, date, newPeriod);
+			this._setDateField(periodName, date, newPeriod);
+			this._editorInput.val(this._getDisplayValue());
+			if (userInteraction) {
```

Comment:
Actually, moving value processing (`_triggerInternalValueChange`) to optional step later can leave `_getDisplayValue()` applying the current value before the spin (e.g. empty initial value will get updated to today, but the display will be left blank).